semantic-ast/src/Main.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE ApplicativeDo #-}
+{-# LANGUAGE TypeApplications #-}
+module Main (main) where
+
+import System.Environment
+import TreeSitter.Unmarshal
+import TreeSitter.Python.AST
+import TreeSitter.Python
+import Source.Range
+import Source.Span
+import Data.ByteString.Char8
+import Data.ByteString (pack, readFile, ByteString)
+import System.IO (FilePath)
+import Options.Applicative hiding (style)
+import Data.Semigroup ((<>))
+
+data SemanticAST = SemanticAST
+  { format              :: Format
+  , source              :: Either FilePath Prelude.String
+  } deriving (Read)
```

Comment:
I don’t think you need a `Read` constraint here any more (if you do, it might be a bug, or at least something suboptimal).